File: tests/human/x/fs/load_yaml.fs
open System
open System.IO
open YamlDotNet.Serialization

[<CLIMutable>]
type Person = { name: string; age: int; email: string }

let deserializer = DeserializerBuilder().Build()
let yamlText = File.ReadAllText("../../interpreter/valid/people.yaml")
let people = deserializer.Deserialize<Person list>(yamlText)

let adults =
    people
    |> List.filter (fun p -> p.age >= 18)
    |> List.map (fun p -> {| name = p.name; email = p.email |})

for a in adults do
    printfn "%s %s" a.name a.email

File: tests/human/x/clj/load_yaml.clj
(ns load-yaml
  (:require [clojure.string :as str]
            [clojure.java.io :as io]))

(defn parse-people [path]
  (->> (str/split-lines (slurp path))
       (partition 3)
       (map (fn [[n a e]]
              {:name (str/trim (subs n 7))
               :age (Integer/parseInt (str/trim (subs a 6)))
               :email (str/trim (subs e 8))}))))

(def people (parse-people "../interpreter/valid/people.yaml"))

(def adults (filter #(>= (:age %) 18) people))
(doseq [a adults]
  (println (:name a) (:email a)))
